kernel/src/schema.rs

```diff
@@ -29,6 +29,17 @@ pub enum MetadataValue {
     Other(serde_json::Value),
 }
 
+impl MetadataValue {
+    pub(crate) fn as_string(&self) -> String {
+        match self {
+            MetadataValue::Number(n) => format!("{}", n),
```

Comment:
aside: that `i32` in `Number` is going to burn us some day... can we change it to `i64`?